modules/core/arrow-annotations-processor/src/main/java/arrow/common/utils/ProcessorUtils.kt

```diff
@@ -47,13 +47,14 @@ interface ProcessorUtils : KotlinMetadataUtils {
   }
 
   fun getClassOrPackageDataWrapper(classElement: TypeElement): ClassOrPackageDataWrapper {
-    val metadata = classElement.kotlinMetadata ?: knownError("These annotations can only be used in Kotlin")
+    val metadata = classElement.kotlinMetadata ?: knownError("Arrow's annotations can only be used on Kotlin classes. Not valid for $classElement")
+
     return metadata.asClassOrPackageDataWrapper(classElement)
-      ?: knownError("This annotation can't be used on this element")
+      ?: knownError("Arrow's annotation can't be used on $classElement")
```

Comment:
can't you get the potential binary name for this element (class name) from the `TypeElement` to include it in the error instead of the TypeElement instance reference? (`ProcessingEnvironment.getElementUtils().getBinaryName(TypeElement)`).

Reply:
Actually for

```kotlin
@instance(String::class)
interface StringEqInstance : Eq<String> {
  override fun String.eqv(b: String): Boolean = this == b
}
```

it prints `e: error: Arrow's annotations can only be used on Kotlin classes. Not valid for java.lang.String`. Not sure if that's what you wanted?

`toString()` is correctly implemented for `Element` and it thus prints the class names.

Reply:
no, I was expecting to get the same but for the annotated class itself, like `arrow.StringEqInstance`. Are you calling it for the `classElement` argument?

Reply:
Yes, that makes sense. We need to improve error message.

I tried to write an `Each` instance for `String` and got `This annotation can't be used on this element`. So I made a small change to improve what we currently have.

We should, however, improve error message for all processors but that doesn't belong in this PR. I would go as far as provide clear custom messages for each error case.

In this case something like.

```
e: error: @instance cannot be used for java.lang.String (or even better kotlin.String)

interface StringEqInstance : Eq<String>
                  ^

Cannot generate instance method for arrow.data.instances.StringEqInstance.
```